cmd/faucet/faucet.go

```diff
@@ -176,107 +273,33 @@ func main() {
 	var genesis *genesisT.Genesis
 	var enodes []*discv5.Node
 	var blob []byte
 
-	genesis, *bootFlag, *netFlag = func() (gs *genesisT.Genesis, bs string, netid uint64) {
-		var configs = []struct {
-			flag bool
-			gs   *genesisT.Genesis
-			bs   []string
-		}{
-			{
-				*foundationFlag,
-				params.DefaultGenesisBlock(),
-				nil,
-			},
-			{
-				*classicFlag,
-				params.DefaultClassicGenesisBlock(),
-				nil,
-			},
-			{
-				*mordorFlag,
-				params.DefaultMordorGenesisBlock(),
-				nil,
-			},
-			{
-				*socialFlag,
-				params.DefaultSocialGenesisBlock(),
-				params.SocialBootnodes,
-			},
-			{
-				*ethersocialFlag,
-				params.DefaultEthersocialGenesisBlock(),
-				params.EthersocialBootnodes,
-			},
-			{
-				*mixFlag,
-				params.DefaultMixGenesisBlock(),
-				params.MixBootnodes,
-			},
-			{
-				*testnetFlag,
-				params.DefaultRopstenGenesisBlock(),
-				nil,
-			},
-			{
-				*rinkebyFlag,
-				params.DefaultRinkebyGenesisBlock(),
-				nil,
-			},
-			{
-				*kottiFlag,
-				params.DefaultKottiGenesisBlock(),
-				nil,
-			},
-			{
-				*goerliFlag,
-				params.DefaultGoerliGenesisBlock(),
-				nil,
-			},
-		}
-
-		var bss []string
-		for _, conf := range configs {
-			if conf.flag {
-				gs, bss, netid = conf.gs, conf.bs, *conf.gs.Config.GetNetworkID()
-				break
-			}
-		}
-		if len(bss) > 0 {
-			bs = strings.Join(bss, ",")
-		}
+	// client will be used if the faucet is attaching. If not it won't be touched.
+	var client *ethclient.Client
 
-		// allow overrides
-		if *genesisFlag != "" {
-			blob, err = ioutil.ReadFile(*genesisFlag)
-			if err != nil {
-				log.Crit("Failed to read genesis block contents", "genesis", *genesisFlag, "err", err)
-			}
-			gs = new(genesisT.Genesis)
-			if err = json.Unmarshal(blob, gs); err != nil {
-				log.Crit("Failed to parse genesis block json", "err", err)
+	genesis, *bootFlag, *netFlag = parseChainFlags()
+
+	if genesis != nil {
+		log.Info("Using chain/net config", "network id", *netFlag, "bootnodes", *bootFlag, "chain config", fmt.Sprintf("%v", genesis.Config))
+
+		// Convert the bootnodes to internal enode representations
+		for _, boot := range utils.SplitAndTrim(*bootFlag) {
+			if url, err := discv5.ParseNode(boot); err == nil {
+				enodes = append(enodes, url)
+			} else {
+				log.Error("Failed to parse bootnode URL", "url", boot, "err", err)
 			}
 		}
-		if *bootFlag != "" {
-			bs = *bootFlag
-		}
-		if *netFlag != 0 {
-			netid = *netFlag
-		}
-		return
-	}()
-	log.Info("configured chain/net config", "network id", *netFlag, "bootnodes", *bootFlag, "chain config", fmt.Sprintf("%v", genesis.Config))
-
-	// Convert the bootnodes to internal enode representations
-	for _, boot := range utils.SplitAndTrim(*bootFlag) {
-		if url, err := discv5.ParseNode(boot); err == nil {
-			enodes = append(enodes, url)
-		} else {
-			log.Error("Failed to parse bootnode URL", "url", boot, "err", err)
+	} else {
+		log.Info("Attaching faucet to running client")
```

Comment:
@meowsbits this message is a bit misleading. While testing #268 without using neither `-attach` nor a chain `-chain.*` flag. It get's in here saying it failed to attach. While the truth is that it failed to peek a chain.

Reply:
Nice catch. Please see https://github.com/etclabscore/core-geth/pull/275.